cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,341 @@
+			}
+			userArn := raw.UserIdentity.SessionContext.SessionIssuer.Arn
+			regexOdj := regexp.MustCompile(mergedRegex)
+			matchesUsername := false
+			matchesArn := false
+
+			if !allEvents && len(Ignore) != 0 {
+				if event.Username != nil {
+					matchesUsername = regexOdj.MatchString(*event.Username)
+				}
+				if userArn != "" {
+					matchesArn = regexOdj.MatchString(userArn)
+
+				}
+
+				if matchesUsername || matchesArn {
+					continue
+					// skips entry
+				}
+
+			}
+			filteredEvents = append(filteredEvents, event)
+
+		}
+	}
+
+	return &filteredEvents, nil
+}
+
+// PrintEvents prints the details of each event in the provided slice of events.
+// It takes a slice of types.Event
+func printEvents(filteredEvent []types.Event, printUrl bool, raw bool) {
+	for _, event := range filteredEvent {
+		if raw {
+			if event.CloudTrailEvent != nil {
+				fmt.Printf("%v \n\n", *event.CloudTrailEvent)
+			}
+		} else {
+			rawEventDetails, err := extractUserDetails(event.CloudTrailEvent)
+			if err != nil {
+				fmt.Printf("[Error] Error extracting event details: %v", err)
+			}
+			sessionIssuer := rawEventDetails.UserIdentity.SessionContext.SessionIssuer.UserName
+			if event.EventName != nil {
+				fmt.Printf("%v ", *event.EventName)
+			} else {
+				continue
+			}
+
+			if event.EventTime != nil {
+				fmt.Printf("|%v ", event.EventTime.String())
+				if event.Username != nil {
+					fmt.Printf("| User: %v ", *event.Username)
+				}
+				if sessionIssuer != "" {
+					fmt.Printf("| ARN: %v\n", sessionIssuer)
+				} else {
+					fmt.Print("\n")
+				}
+
+				if printUrl && event.CloudTrailEvent != nil {
+					if err == nil {
+						fmt.Printf("EventLink: %v\n\n", generateLink(*rawEventDetails))
+					} else {
+						fmt.Println("EventLink: <not available>")
+					}
+				}
+
+			}
+		}
+	}
+
+}
+
+func (o *LookupEventsOptions) run() error {
+
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return fmt.Errorf("unable to create connection to ocm: %w", err)
+	}
+	defer connection.Close()
+
+	cluster, err := utils.GetClusterAnyStatus(connection, o.clusterID)
+	if err != nil {
+		return err
+	}
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return fmt.Errorf("[ERROR] this command is only available for AWS clusters")
+	}
+
+	Ignore, err := envConfig.LoadCloudTrailConfig()
+	if err != nil {
+		return fmt.Errorf("[ERROR] error Loading cloudtrail configuration file: %w", err)
+	}
+	if len(Ignore) == 0 {
+		fmt.Println("\n[WARNING] No filter list DETECTED!!")
+	}
+
+	cfg, err := osdCloud.CreateAWSV2Config(connection, cluster)
+	if err != nil {
+		return err
+	}
+	DefaultRegion := "us-east-1"
+	configOptions := config.WithRegion(DefaultRegion)
+	defaultConfig, err := config.LoadDefaultConfig(context.Background(), configOptions)
+	if err != nil {
+		return err
+	}
```

Comment:
We should only touch the `us-east-1` part once we know the cluster's region is different, as commented below, move this to the minimal scope.